python/activator/timer.py

```diff
@@ -0,0 +1,50 @@
+# This file is part of prompt_processing.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["time_this_to_bundle"]
+
+
+from contextlib import contextmanager
+
+import lsst.verify
+
+
+@contextmanager
+def time_this_to_bundle(bundle, action_id, metric):
```

Comment:
Nvm, I now read your comment on jira. Sounds like this will go into `analysis_tools` in the future.

Reply:
Actually, it will probably be removed entirely -- `analysis_tools` keeps the construction of a bundle very well separated from the construction of the metric values that go into it.